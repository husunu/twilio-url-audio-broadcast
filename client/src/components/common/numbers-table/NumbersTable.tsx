import React, { useMemo } from 'react';
import { useNumbersTableStyles, StyledTableCell } from './styles';
import { usePagination } from '@/hooks/usePagination';

import {
  Box,
  TableContainer,
  Table,
  TableHead,
  TableRow,
  TableCell,
  TableBody,
  TablePagination,
} from '@material-ui/core';

export type NumbersTableProps = {
  headers: string[];
  rows: string[];
  data: any[];
};

export const NumbersTable: React.FC<NumbersTableProps> = ({
  headers,
  rows,
  data,
}) => {
  const classes = useNumbersTableStyles();
  const {
    page,
    rowsPerPage,
    handleChangePage,
    handleChangeRowsPerPage,
    paginatedData,
    rowsPerPageOptions,
  } = usePagination(data);

  const rowCells = useMemo(() => {
    return paginatedData.map(
      (rowData: { [key: string]: string }, idx: number) => (
        <TableRow key={idx}>
          {rows.map((cell: string, index: number) => (
            <TableCell key={`${idx}-${index}`}>{rowData[cell]}</TableCell>
          ))}
        </TableRow>
      )
    );
  }, [paginatedData, rows]);

  return (
    <>
      <TableContainer component={Box} className={classes.tableContainer}>
        <Table stickyHeader size="small">
          <TableHead>
            <TableRow>
              {headers.map((title: string, idx: number) => (
                <StyledTableCell key={idx}>{title}</StyledTableCell>
              ))}
            </TableRow>
          </TableHead>
          <TableBody>{rowCells}</TableBody>
        </Table>
      </TableContainer>
      <TablePagination
        rowsPerPageOptions={rowsPerPageOptions}
        component="div"
        count={data.length}
        rowsPerPage={rowsPerPage}
        page={page}
        onChangePage={handleChangePage}
        onChangeRowsPerPage={handleChangeRowsPerPage}
      />
    </>
  );
};
