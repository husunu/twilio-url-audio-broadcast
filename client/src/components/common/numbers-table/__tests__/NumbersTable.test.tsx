import React from 'react';
import { HashRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import { NumbersTable, NumbersTableProps } from '../NumbersTable';

const props: NumbersTableProps = {
  headers: ['Line', 'Numbers', 'Issue'],
  rows: ['line', 'number', 'issue'],
  data: [
    {
      line: 1,
      number: '+14081234567',
      issue: null,
    },
    {
      line: 2,
      number: '+14081234567',
      issue: 'Duplicate Number',
    },
    {
      line: 3,
      number: 'asasd',
      issue: 'Invalid Number',
    },
  ],
};

describe('NumbersTable', () => {
  test('renders the page', () => {
    const { asFragment } = render(
      <Router>
        <NumbersTable {...props} />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
