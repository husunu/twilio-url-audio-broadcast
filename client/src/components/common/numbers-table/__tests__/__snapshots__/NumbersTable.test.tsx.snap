// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumbersTable renders the page 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-4 MuiTableContainer-root makeStyles-tableContainer-1"
  >
    <table
      class="MuiTable-root MuiTable-stickyHeader"
    >
      <thead
        class="MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-13 MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
            scope="col"
          >
            Line
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-13 MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
            scope="col"
          >
            Numbers
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-13 MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
            scope="col"
          >
            Issue
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            1
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            +14081234567
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          />
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            2
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            +14081234567
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            Duplicate Number
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            3
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            asasd
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            Invalid Number
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    class="MuiTablePagination-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
    >
      <div
        class="MuiTablePagination-spacer"
      />
      <p
        class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
      >
        1-3 of 3
      </p>
      <div
        class="MuiTablePagination-actions"
      >
        <button
          aria-label="Previous page"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
          disabled=""
          tabindex="-1"
          title="Previous page"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
              />
            </svg>
          </span>
        </button>
        <button
          aria-label="Next page"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
          disabled=""
          tabindex="-1"
          title="Next page"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
